offchain-modules/packages/x/src/handlers/ckb.ts

```diff
@@ -1032,6 +1076,166 @@ export class CkbHandler {
+
+    const senderTypescript = senderOutput.type;
+    if (
+      senderTypescript &&
+      (senderTypescript.codeHash !== sudtTypescript.codeHash || senderTypescript.hashType !== sudtTypescript.hashType)
+    ) {
+      logger.warn(`invalid unlock tx: the typescript of first input is not null or sudt typescript`);
+      return null;
+    }
+
+    for (let i = 1; i < tx.inputs.length; i++) {
+      const thisPreviousOutput = nonNullable(tx.inputs[i].previousOutput);
+      const thisPreHash = thisPreviousOutput.txHash;
+      const thisTxPrevious = await this.ckb.rpc.getTransaction(thisPreHash);
+      if (thisTxPrevious == null) {
+        return null;
+      }
+      const thisSenderLockscript = thisTxPrevious.transaction.outputs[Number(thisPreviousOutput.index)].lock;
+      if (
+        !thisSenderLockscript ||
+        !(
+          (thisSenderLockscript.codeHash !== committeeMultisigLockscript.code_hash &&
+            thisSenderLockscript.hashType !== committeeMultisigLockscript.hash_type &&
+            thisSenderLockscript.args !== committeeMultisigLockscript.args) ||
+          (thisSenderLockscript.codeHash !== collectorLockscript.code_hash &&
+            thisSenderLockscript.hashType !== collectorLockscript.hash_type &&
+            thisSenderLockscript.args !== collectorLockscript.args)
+        )
+      ) {
+        logger.warn(`invalid unlock tx: the lockscript of inputs must be committee multisig or collector cell`);
+        return null;
+      }
+    }
+
+    if (tx.inputs.length + 1 !== tx.witnesses.length) {
+      logger.warn(`invalid unlock tx: the length of witness should be inputs.length + 1`);
+      return null;
+    }
+    const unlockMemoWitness = tx.witnesses[tx.witnesses.length - 1];
+    let unlockMemo: UnlockMemo;
+    try {
+      unlockMemo = new UnlockMemo(new Reader(unlockMemoWitness).toArrayBuffer());
+    } catch (e) {
+      logger.error(
+        `invalid unlock tx: parse recipient unlock memo in witness error: ${e.message} ${e.stack}, tx: ${tx}`,
+      );
+      return null;
+    }
+    const xchain = unlockMemo.getXchain();
+    const burnIdBytes: BytesVec = unlockMemo.getBurnIds();
+    const burnIds = lodash
+      .range(burnIdBytes.length())
+      .map((i) => new Reader(burnIdBytes.indexAt(i).raw()).serializeJson());
+
+    if (burnIds.length > tx.outputs.length - 2) {
+      logger.warn(
+        `invalid unlock tx: burnIds in unlock memo are more than outputs: burnIds: ${burnIds}, outputs: ${tx.outputs}`,
+      );
+      return null;
+    }
+    const isCkb = !senderTypescript;
+    const assetIdent = isCkb
+      ? '0x0000000000000000000000000000000000000000000000000000000000000000'
+      : utils.computeScriptHash({
+          code_hash: senderTypescript!.codeHash,
+          hash_type: senderTypescript!.hashType,
+          args: senderTypescript!.args,
+        });
+
+    const iCkbUnlocks: ICkbUnlock[] = [];
+    const ethBurns: EthBurn[] = await this.db.getEthBurnByUniqueIds(burnIds);
```

Comment:
Don't get data from EthBurn, EthBurn record maybe null. We just get unlock_related_data from tx outputs, and save it to db

Reply:
fixed, get burnTxHash by splite from burnId